So I tried that I am getting:
```
Error - <class 'subprocess.CalledProcessError'>: Command '['df', '-h', './database/files']' returned non-zero exit status 1.
URL: http://localhost:9001/system
File 'lib/galaxy/web/framework/middleware/error.py', line 153 in __call__
  app_iter = self.application(environ, sr_checker)
File '/home/oleg/galaxy/.venv/lib/python3.8/site-packages/paste/recursive.py', line 85 in __call__
  return self.application(environ, start_response)
File '/home/oleg/galaxy/.venv/lib/python3.8/site-packages/paste/httpexceptions.py', line 640 in __call__
  return self.application(environ, start_response)
File 'lib/galaxy/web/framework/base.py', line 141 in __call__
  return self.handle_request(environ, start_response)
File 'lib/galaxy/web/framework/base.py', line 220 in handle_request
  body = method(trans, **kwargs)
File 'lib/galaxy/webapps/reports/controllers/system.py', line 34 in index
  file_path, disk_usage, datasets, file_size_str = self.disk_usage(trans, **kwd)
File 'lib/galaxy/webapps/reports/controllers/system.py', line 185 in disk_usage
  disk_usage = self.get_disk_usage(file_path)
File 'lib/galaxy/webapps/reports/controllers/system.py', line 155 in get_disk_usage
  df_output = unicodify(subprocess.check_output(['df', '-h', file_path]))
File '/usr/lib/python3.8/subprocess.py', line 411 in check_output
  return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
File '/usr/lib/python3.8/subprocess.py', line 512 in run
  raise CalledProcessError(retcode, process.args,
CalledProcessError: Command '['df', '-h', './database/files']' returned non-zero exit status 1.


CGI Variables
-------------
  HTTP_ACCEPT: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9'
  HTTP_ACCEPT_ENCODING: 'gzip, deflate, br'
  HTTP_ACCEPT_LANGUAGE: 'en-US,en;q=0.9,ru;q=0.8,de;q=0.7'
  HTTP_CONNECTION: 'keep-alive'
  HTTP_COOKIE: 'Pycharm-63826326=66f368d2-bdc6-4ff3-a565-8ce3791e87e2; galaxysession=c6ca0ddb55be603a79551a01b14148feb15d5e10d576074cd108baf535b635da62518c1b5548d293'
  HTTP_HOST: 'localhost:9001'
  HTTP_REFERER: 'http://localhost:9001/'
  HTTP_SEC_FETCH_DEST: 'iframe'
  HTTP_SEC_FETCH_MODE: 'navigate'
  HTTP_SEC_FETCH_SITE: 'same-origin'
  HTTP_SEC_FETCH_USER: '?1'
  HTTP_UPGRADE_INSECURE_REQUESTS: '1'
  HTTP_USER_AGENT: 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.121 Safari/537.36'
  PATH_INFO: '/system'
  REMOTE_ADDR: '127.0.0.1'
  REMOTE_PORT: '31360'
  REQUEST_METHOD: 'GET'
  REQUEST_URI: '/system'
  SERVER_NAME: 'oleg-laptop'
  SERVER_PORT: '9001'
  SERVER_PROTOCOL: 'HTTP/1.1'
  UWSGI_ROUTER: 'http'


WSGI Variables
--------------
  application: <paste.recursive.RecursiveMiddleware object at 0x7f1db900b8e0>
  controller_action_key: 'web.system.index'
  is_api_request: False
  paste.expected_exceptions: [<class 'paste.httpexceptions.HTTPException'>]
  paste.httpexceptions: <paste.httpexceptions.HTTPExceptionHandler object at 0x7f1db900baf0>
  paste.recursive.forward: <paste.recursive.Forwarder from />
  paste.recursive.include: <paste.recursive.Includer from />
  paste.recursive.include_app_iter: <paste.recursive.IncluderAppIter from />
  paste.recursive.script_name: ''
  paste.throw_errors: True
  uwsgi.core: 3
  uwsgi.node: b'oleg-laptop'
  uwsgi.version: b'2.0.19.1'
  webob._parsed_query_vars: (GET([]), '')
  wsgi process: 'Multithreaded'
  wsgi.file_wrapper: <built-in function uwsgi_sendfile>
------------------------------------------------------------
[ERROR] Unhandled object from iterator: '\n    <!DOCTYPE HTML>\n    <html>\n    <head>\n    <style type="text/css">\n    body { color: #303030; background: #dfe5f9; font-family:"Lucida Grande",verdana,arial,helvetica,sans-serif; font-size:12px; line-height:16px; }\n    .content { max-width: 720px; margin: auto; margin-top: 50px; }\n    </style>\n    <title>Internal Server Error</title>\n    \n    </head>\n    <body>\n    <div class="content">\n    <h1>Internal Server Error</h1>\n\n    <h2>Galaxy was unable to successfully complete your request</h2>\n\n    <p>\n            An error occurred.\n            </p>\n\n    This may be an intermittent problem due to load or other unpredictable factors, reloading the page may address the problem.\n\n    <p><b>The error has been logged to our team.</b></p>\n    </div>\n    </body>\n    </html>' (0x7f1dac08c750)
```

--- a/lib/galaxy/managers/libraries.py
+++ b/lib/galaxy/managers/libraries.py
@@ -3,7 +3,7 @@ Manager and Serializer for libraries.
 """
 import logging
 
-from sqlalchemy import false, not_, or_, true
+from sqlalchemy import and_, false, not_, or_, true
 from sqlalchemy.orm.exc import MultipleResultsFound
 from sqlalchemy.orm.exc import NoResultFound
 
@@ -292,3 +292,18 @@ class LibraryManager:
         Return true if lib is public.
         """
         return trans.app.security_agent.library_is_public(library)
+
+
+def get_containing_library_from_library_dataset(trans, library_dataset):
+    """Given a library_dataset, get the containing library"""
+    folder = library_dataset.folder
+    while folder.parent:
+        folder = folder.parent
+    # We have folder set to the library's root folder, which has the same name as the library
+    for library in trans.sa_session.query(trans.model.Library).filter(
+        and_(trans.model.Library.table.c.deleted == false(),
+            trans.model.Library.table.c.name == folder.name)):
+        # Just to double-check
+        if library.root_folder == folder:
+            return library
+    return None

--- a/lib/galaxy/webapps/reports/controllers/system.py
+++ b/lib/galaxy/webapps/reports/controllers/system.py
@@ -1,6 +1,6 @@
 import logging
 import os
-import subprocess
+import shutil
 from datetime import datetime, timedelta
 from decimal import Decimal
 
@@ -150,34 +150,9 @@ class System(BaseUIController):
                                    message=message)
 
     def get_disk_usage(self, file_path):
-        is_sym_link = os.path.islink(file_path)
-        file_system = disk_size = disk_used = disk_avail = disk_cap_pct = mount = None
-        df_output = unicodify(subprocess.check_output(['df', '-h', file_path]))
-
-        for df_line in df_output.splitlines():
-            df_line = df_line.strip()
-            if df_line:
-                df_line = df_line.lower()
-                if 'filesystem' in df_line or 'proc' in df_line:
-                    continue
-                elif is_sym_link:
-                    if ':' in df_line and '/' in df_line:
-                        mount = df_line
-                    else:
-                        try:
-                            disk_size, disk_used, disk_avail, disk_cap_pct, file_system = df_line.split()
-                            break
-                        except Exception:
-                            pass
-                else:
-                    try:
-                        file_system, disk_size, disk_used, disk_avail, disk_cap_pct, mount = df_line.split()
-                        break
-                    except Exception:
-                        pass
-            else:
-                break  # EOF
-        return (file_system, disk_size, disk_used, disk_avail, disk_cap_pct, mount)
+        disk_usage = shutil.disk_usage(file_path)
+        pct_used = round(disk_usage.used / disk_usage.total * 100, 2)
+        return (nice_size(disk_usage.total), nice_size(disk_usage.used), nice_size(disk_usage.free), pct_used)
 
     @web.expose
     def disk_usage(self, trans, **kwd):

--- a/lib/galaxy/webapps/reports/controllers/system.py
+++ b/lib/galaxy/webapps/reports/controllers/system.py
@@ -1,5 +1,4 @@
 import logging
-import os
 import shutil
 from datetime import datetime, timedelta
 from decimal import Decimal
@@ -8,7 +7,6 @@ from sqlalchemy import and_, desc, false, null, true
 from sqlalchemy.orm import eagerload
 
 from galaxy import model, util
-from galaxy.util import unicodify
 from galaxy.webapps.base.controller import BaseUIController, web
 
 log = logging.getLogger(__name__)

--- a/lib/galaxy/webapps/reports/controllers/tools.py
+++ b/lib/galaxy/webapps/reports/controllers/tools.py
@@ -302,7 +302,7 @@ class Tools(BaseUIController):
         if tool_name is None:
             raise ValueError("Tool can't be none")
         tool_errors = [[unicodify(a), b] for a, b in
-                       sa.select((galaxy.model.Job.table.c.stderr, galaxy.model.Job.table.c.create_time),
+                       sa.select((galaxy.model.Job.table.c.tool_stderr, galaxy.model.Job.table.c.create_time),
                         from_obj=[galaxy.model.Job.table],
                         whereclause=and_(galaxy.model.Job.table.c.tool_id == tool_name,
                                          galaxy.model.Job.table.c.state == 'error')).execute()]

--- a/templates/webapps/reports/dataset_info.mako
+++ b/templates/webapps/reports/dataset_info.mako
@@ -3,7 +3,7 @@
 
 <%
     from galaxy.web.framework.helpers import time_ago
-    from galaxy.webapps.galaxy.controllers.library_common import get_containing_library_from_library_dataset
+    from galaxy.managers.libraries import get_containing_library_from_library_dataset
 %>
 
 %if message:

--- a/templates/webapps/reports/system.mako
+++ b/templates/webapps/reports/system.mako
@@ -62,7 +62,6 @@
             </td>
         </tr>
         <tr class="header">
-            <td>File System</td>
             <td>Disk Size</td>
             <td>Used</td>
             <td>Available</td>
@@ -73,7 +72,6 @@
             <td>${disk_usage[1]}</td>
             <td>${disk_usage[2]}</td>
             <td>${disk_usage[3]}</td>
-            <td>${disk_usage[4]}</td>
         </tr>
     </table>
     <br clear="left" />

